internal/services/graph/graph_account_resource_test.go

```diff
@@ -0,0 +1,161 @@
+	data := acceptance.BuildTestData(t, "azurerm_graph_account", "test")
+	r := AccountTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccGraphAccount_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_graph_account", "test")
+	r := AccountTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+func (r AccountTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := graphservicesprods.ParseAccountID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Graph.V20230413.Graphservicesprods.AccountsGet(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+func (r AccountTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_graph_account" "test" {
+  name                = "acctesta-%[2]d"
+  app_id              = azuread_application.test.application_id
+  resource_group_name = azurerm_resource_group.test.name
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r AccountTestResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_graph_account" "import" {
+  app_id              = azurerm_graph_account.test.app_id
+  name                = azurerm_graph_account.test.name
+  resource_group_name = azurerm_graph_account.test.resource_group_name
+}
+`, r.basic(data))
+}
+
+func (r AccountTestResource) complete(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_graph_account" "test" {
+  name                = "acctesta-%[2]d"
+  resource_group_name = azurerm_resource_group.test.name
+  app_id              = azuread_application.test.application_id
+  tags = {
+    environment = "terraform-acctests"
+    some_key    = "some-value"
+  }
+}
+`, r.template(data), data.RandomInteger)
+}
+
+func (r AccountTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azuread" {}
+
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestrg-graph-%d"
+  location = %q
+}
+
+resource "azuread_application" "test" {
```

Comment:
this resource is depend on an `azuread_application`, however through my local test the resource creation may fail occationally with error: `performing AccountsCreateAndUpdate: unexpected status 400 with error: BadRequest: The application was not found in this tenant.`. The service team told that this is hard to change and advices to use a predefined azure ad app, can we do this?